frontend/src/metabase/lib/entities.js

```diff
@@ -30,160 +30,10 @@ import _ from "underscore";
-    [name: string]: Function,
-  },
-  objectActions?: {
-    [name: string]: ObjectActionCreator,
-  },
-  objectSelectors?: {
-    [name: string]: ObjectSelector,
-  },
-  reducer?: Reducer,
-  wrapEntity?: (object: EntityObject) => any,
-  actionShouldInvalidateLists?: (action: Action) => boolean,
-
-  // list of properties for this object which should be persisted
-  writableProperties?: string[],
-};
-
-type EntityObject = any;
-
-type EntityQuery = {
-  [name: string]: string | number | boolean | null,
-};
-
-type FetchOptions = {
-  reload?: boolean,
-};
-type UpdateOptions = {
-  notify?:
-    | { verb?: string, subject?: string, undo?: boolean, message?: any }
-    | false,
-};
-
-type Result = any; // FIXME
-
-export type Entity = {
-  name: EntityName,
-
-  nameOne: string,
-  nameMany: string,
-
-  displayNameOne: string,
-  displayNameMany: string,
-
-  form?: FormDefinition,
-  forms?: { [key: string]: FormDefinition },
-
-  path?: string,
-  api: {
-    list: APIMethod,
-    create: APIMethod,
-    get: APIMethod,
-    update: APIMethod,
-    delete: APIMethod,
-    [method: string]: APIMethod,
-  },
-  schema: schema.Entity,
-  actionTypes: {
-    [name: string]: ActionType,
-    CREATE: ActionType,
-    FETCH: ActionType,
-    UPDATE: ActionType,
-    DELETE: ActionType,
-    FETCH_LIST: ActionType,
-  },
-  actionDecorators: {
-    [name: string]: Function, // TODO: better type
-  },
-  actions: {
-    [name: string]: ActionCreator,
-    fetchList: (
-      entityQuery?: EntityQuery,
-      options?: FetchOptions,
-    ) => Promise<Result>,
-  },
-  reducers: { [name: string]: Reducer },
-  selectors: {
-    getList: Function,
-    getObject: Function,
-    getLoading: Function,
-    getLoaded: Function,
-    getFetched: Function,
-    getError: Function,
-    [name: string]: Function,
-  },
-  objectActions: {
-    [name: string]: ObjectActionCreator,
-    create: (entityObject: EntityObject) => Promise<Result>,
-    fetch: (
-      entityObject: EntityObject,
-      options?: FetchOptions,
-    ) => Promise<Result>,
-    update: (
-      entityObject: EntityObject,
-      updatedObject: EntityObject,
-      options?: UpdateOptions,
-    ) => Promise<Result>,
-    delete: (entityObject: EntityObject) => Promise<Result>,
-  },
-  objectSelectors: {
-    [name: string]: ObjectSelector,
-  },
-  wrapEntity: (object: EntityObject) => any,
-
-  requestsReducer: Reducer,
-  actionShouldInvalidateLists: (action: Action) => boolean,
-
-  writableProperties?: string[],
-  getAnalyticsMetadata?: () => any,
-
-  normalize: (object: EntityObject, schema?: schema.Entity) => any, // FIXME: return type
-  normalizeList: (list: EntityObject[], schema?: schema.Entity) => any, // FIXME: return type
-
-  getObjectStatePath: Function,
-  getListStatePath: Function,
-
-  HACK_getObjectFromAction: (action: Action) => any,
-};
-
-export function createEntity(def: EntityDefinition): Entity {
-  const entity: Entity = { ...def };
+// FIXME
```

Comment:
let's remove this comment

Reply:
Sure, I'll remove them in another open PR